weldx/util.py

```diff
@@ -1222,12 +1241,19 @@ def _visit(p, k, v, b):
         2. If the index `k` does not exist an IndexError is raised.
         3. If the other path exists, a comparison will be made using `_compare`.
            When the elements are not equal traversing `a` will be stopped
-           by raising a ValueError.
+           by raising a RuntimeError.
         """
-        other_value = iterutils.get_path(b, p)[k]
-        if not iterutils.default_enter(p, k, v)[1]:
-            if not _Eq_compare_nested._compare(v, other_value):
-                raise ValueError(f"{v=}\n{other_value=}")
+        other_data_structure = iterutils.get_path(b, path)
+        other_value = other_data_structure[key]
+        if not _Eq_compare_nested._enter(None, key, value)[1]:
+            # check lengths of Sequence types first and raise prior starting a more expensive comparison!
+            if isinstance(other_data_structure, Sequence) and len(
+                other_data_structure
+            ) != len(iterutils.get_path(a, path)):
+                raise RuntimeError
+
+            if not _Eq_compare_nested._compare(value, other_value):
+                raise RuntimeError
```

Comment:
Even if it doesn't show up, can you add a small description of the error case in the `RuntimeError` here?

Reply:
ok, but I honestly do not see why it would be useful :)